### PR TITLE
Adds hover functionality

### PR DESCRIPTION
**Pull request by nehau9**
Implemented the hovertext() functionality in plot_colored_table in visualize_benchmark_comparison.py. It now displays the value in the master followed by the value in the branch and the percentage change w.r.t. value in master.

### CODE REVIEW

File: gtsfm/evaluation/visualize_benchmark_comparison.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def colorscale_from_list(colorlist: List[str]) -> List[str]:
     return colorscale
 
 
-def plot_colored_table(row_labels: List[str], col_labels: List[str], tab_data: np.ndarray) -> str:
+def plot_colored_table(new_values: np.ndarray, old_values: np.ndarray, row_labels: List[str], col_labels: List[str], tab_data: np.ndarray) -> str:
```
Comment (johnwlambert) [inline gtsfm/evaluation/visualize_benchmark_comparison.py:67]: looks good @nehau9.

This line is a bit long (147 chars) -- can you reformat the file with 120-char max length?
`black gtsfm/evaluation/visualize_benchmark_comparison.py -l 120`
after `pip install black`
Thanks!

Comment (johnwlambert) [reply gtsfm/evaluation/visualize_benchmark_comparison.py:67]: could you add info about `new_values` and `old_values` to the docstring `Args` also?

File: gtsfm/evaluation/visualize_benchmark_comparison.py [hunk 2]
```diff
@@ -76,9 +76,15 @@ def plot_colored_table(row_labels: List[str], col_labels: List[str], tab_data: n
...
-
+    if tab_data.size == 0:
+        return ''
     # Clip "Z" to -20% and +20%. The clipping is only for the color -- the text will still display the correct numbers.
     tab_data_clipped = np.clip(tab_data, a_min=MIN_RENDERABLE_PERCENT_CHANGE, a_max=MAX_RENDERABLE_PERCENT_CHANGE)
+    hovertext = list()
+    for yi, yy in enumerate(col_labels):
+        hovertext.append(list())
+        for xi, xx in enumerate(row_labels):
+            hovertext[-1].append('Master_val: {}<br />Branch_val: {}<br />Percentage: {}'.format(old_values[yi][xi],new_values[yi][xi], tab_data[yi][xi]))
```
Comment (johnwlambert) [inline gtsfm/evaluation/visualize_benchmark_comparison.py:87]: If we're filling up a 2d table, I prefer we make it a bit more explicit, e.g.:
```python
>>> np.zeros((3,5), dtype=str).tolist()
[['', '', '', '', ''], ['', '', '', '', ''], ['', '', '', '', '']]
```
in our case,
```python
H, W = tab_data.shape
hovertext_table = np.zeros((H,W), dtype=str)
for r in range(H):
      for c in range(W):
             cell_text = f"Master: {old_values[r,c]}<br />Branch: {new_values[r,c} <br />Percentage: {tab_data[r,c}"
             hovertext_table[r, c] = cell_text
```
I'd prefer we use f-strings for readability, too.
Should the hover text table argument for Plotly be indexed by (y,x) or (x,y)?

Also curious how you found the documentation for this! Was there a page that was particularly helpful?

**Comment by johnwlambert** [comment]
Example file:
[visual_comparison_dashboard.html.zip](https://github.com/borglab/gtsfm/files/7680930/visual_comparison_dashboard.html.zip)

**Review by johnwlambert — APPROVED** [review]

**Comment by johnwlambert** [comment]
Diff vs. master:
[visual_comparison_dashboard.html.zip](https://github.com/borglab/gtsfm/files/7681963/visual_comparison_dashboard.html.zip)

**Review by travisdriver — APPROVED** [review]
LGTM

File: gtsfm/evaluation/visualize_benchmark_comparison.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/evaluation/visualize_benchmark_comparison.py]: i made these updates, should be good to go

File: gtsfm/evaluation/visualize_benchmark_comparison.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/evaluation/visualize_benchmark_comparison.py]: i made these updates, should be good to go

**Pull request merged**